pages/employee/index.tsx

```diff
@@ -1,30 +1,51 @@
-import React, { useContext } from "react"
+import React, { useCallback, useContext, useEffect, useState } from "react"
 import type { NextPage } from "next"
 import { Heading } from "@chakra-ui/layout"
 import ErrorAlert from "@/components/common/ErrorAlert"
 import Loading from "@/components/common/Loading"
 import EmployeeTable from "@/components/table/EmployeeTable"
 import { UserContext } from "@/lib/contexts/FirebaseAuthContext"
 import { useEmployees } from "@/lib/hooks/useList"
+import { ApiGetResponse } from "@/lib/types/hooks"
+import { Employee } from "@/lib/types/apiTypes"
 
 const Employees: NextPage = () => {
     const { user } = useContext(UserContext)
-    const employeesResponse = useEmployees(user)
+    const initialEmployeeResponse = useEmployees(user)
+
+    const [employeesResponse, setEmployeesResponse] = useState<
+        ApiGetResponse<Employee[]>
+    >(initialEmployeeResponse)
+
+    const getLoadedEmployees = useCallback(
```

Comment:
Removing it would create an infinite render loop

Reply:
Hooks has to have all data in dependency array that it's life cycle depends on. In this case it would be the whole initialEmployeeResponse but that value is "changing"/"initialized" every render and that would fire useEffect hook and re-render so we would end up wiith infinite render loop.